test/unittests/LorisForms_Test.php

```diff
@@ -262,6 +470,1773 @@ function testAddElementPassword()
+                " value=\"default_on\" disabled/> text",
+            $this->form->advCheckboxHTML($this->form->form['abc'])
+        );
+    }
+
+    /**
+     * Test that createText creates an element of type text
+     *
+     * @covers LorisForm::createText
+     * @return void
+     */
+    function testCreateText()
+    {
+        $testText = $this->form->createText("abc", "Hello", array());
+        $this->assertEquals("text", $testText['type']);
+        $this->assertEquals("Hello", $testText['label']);
+    }
+
+    /**
+     * Test that createSubmit creates an element of type submit
+     * with a null label and the proper value
+     *
+     * @covers LorisForm::createSubmit
+     * @return void
+     */
+    function testCreateSubmit()
+    {
+        $testSubmit = $this->form->createSubmit("abc", "value1", array());
+        $this->assertEquals("submit", $testSubmit['type']);
+        $this->assertEquals(null, $testSubmit['label']);
+        $this->assertEquals("value1", $testSubmit['value']);
+    }
+
+    /**
+     * Test that createTextArea creates an element of type textarea
+     *
+     * @covers LorisForm::createTextArea
+     * @return void
+     */
+    function testCreateTextArea()
+    {
+        $testText = $this->form->createTextArea("abc", "Hello");
+        $this->assertEquals("textarea", $testText['type']);
+        $this->assertEquals("Hello", $testText['label']);
+    }
+
+    /** 
+     * Test that createSelect creates an element of type 'select'
+     * and that the options attribute is properly set
+     *
+     * @covers LorisForm::createSelect
+     * @return void
+     */
+    function testCreateSelect()
+    {
+        $testSelect = $this->form->createSelect("abc", "Hello", array(), array());
+        $this->assertEquals("select", $testSelect['type']);
+        $this->assertEquals("Hello", $testSelect['label']);
+        $this->assertTrue(isset($testSelect['options']));
+        $this->assertTrue(!isset($testSelect['multiple']));
+    }
+
+    /**
+     * Test that createSelect creates an element of type 'select
+     * and that the 'multiple' attribute is set if specified
+     * 
+     * @covers LorisForm::createSelect
+     * @return void
+     */
+    function testCreateMultiSelect()
+    {
+        $testSelect = $this->form->createSelect(
+            "abc", "Hello", array(), array('multiple' => 'multiple')
+        );
+        $this->assertEquals("select", $testSelect['type']);
+        $this->assertEquals("Hello", $testSelect['label']);
+        $this->assertTrue(isset($testSelect['options']));
+        $this->assertTrue(isset($testSelect['multiple']));
+    }
+
+    /**
+     * Test that isSubmitted returns true/false depending on the element
+     * TODO I am unsure of how submission can be accomplished in the backend
+     *      It might be something only specified in the frontend
+     * 
+     * @covers LorisForm::isSubmitted
+     * @return void
+     */
+    function testIsSubmitted()
```

Comment:
@AlexandraLivadas Ran your tests. I did get the message saying that this test is incomplete. Is this correct?

Reply:
Also, your rebase fixed the issue I was having on Thursday while testing this PR.

Reply:
Yes. If you look at the TODO, it's because I'm pretty sure the submissions are specified from the frontend. If you look at the [method](https://github.com/aces/Loris/blob/20884c793d87b8c63ffafbaedb01eb650df4e3b3/php/libraries/LorisForm.class.inc#L1192), I can't seem to find where those arrays FILES and POST are declared and/or changed.

Reply:
@kongtiaowang Can you confirm this?

Reply:
@AlexandraLivadas 
please try this to complete this method.
    function testIsSubmitted()
    {
        $_POST = "test";
        $this->assertTrue($this->form->isSubmitted());
        unset($_POST);
        $this->assertFalse($this->form->isSubmitted());
    }